src/fetchPost.js

```diff
@@ -0,0 +1,365 @@
+        infobox = deepSearchByPair(data, "widget_type", "GROUP_INFO_ROW")[0][
+          "data"
+        ]["items"];
+      } catch (error) {}
+
+      let rows = deepSearchByPair(data, "widget_type", "UNEXPANDABLE_ROW");
+      let description = deepSearchByPair(
+        data,
+        "widget_type",
+        "DESCRIPTION_ROW"
+      )[0]["data"]["text"];
+      description = description.replaceAll("\n", "<br/>");
+
+      let hasImage =
+        deepSearchByPair(data, "widget_type", "IMAGE_CAROUSEL").length > 0;
+      let images = hasImage
+        ? deepSearchByPair(data, "widget_type", "IMAGE_CAROUSEL")[0]["data"][
+            "items"
+          ]
+        : [];
+
+      let container = document.querySelector(".single-post-container");
+      container.innerHTML = `
+      <main class="single-post">
+        <nav class="single-post__breadcrumb">
+          ${breadcrumb
+            .map((item) => {
+              return `
+          <span class="single-post__arrowed-text"> ${
+            item.title
+          } <img class="single-post__arrow"src="${
+                require("./icons/svgs/chevron-left.svg").default
+              }"/>
+          </span>`;
+            })
+            .join("")}
+        </nav>
+        <article class="single-post__main-content">
+        <section class="single-post__right-box">
+          <h1 class="single-post__title">
+            ${title}
+          </h1>
+          <section class="single-post__author">
+            ${author}
+          </section>
+          <section class="single-post__actions">
+            <button id="contact-info" class="single-post__action-button button button--primary">
+              <span class="button__title"> اطلاعات تماس </span>
+            </button>
+            <button class="single-post__action-button button button--secondary">
+              <span class="button__title"> چت </span>
+            </button>
+            <button class="single-post__action-bookmark ${
+              bookmarked ? "bookmarked " : ""
+            }button button--icon">
+              <img class="single-post__bookmark-icon" src="${
+                bookmarked
+                  ? require("./icons/svgs/bookmarked.svg").default
+                  : require("./icons/svgs/bookmark-o.svg").default
+              }"/>
+            </button>
+            <button class="single-post__action-share button button--icon">
+              <img class="single-post__share-icon" src="${
+                require("./icons/svgs/share-variant-o.svg").default
+              }"/>
+            </button>
+          </section>
+          <section class="single-post__contact-info-container height-0">
+          </section>
+          ${
+            infobox !== null
+              ? `<section class="single-post__info-box-container">
+          ${infobox
+            .map(
+              (item) => `
+          <section class="single-post__info-box">
+            <span class="single-post__info-box-title"> ${item.title} </span>
```

Comment:
Same pseudo block here, I'm not gonna comment on the rest of the HTML, but you need to read them again yourself and check for these issues.